util/reggen/validate.py

```diff
@@ -1597,27 +1615,38 @@ def validate(regs, **kwargs):
     # auto header generation would go here and update autoregs
 
     if 'no_auto_intr_regs' in regs:
-        no_autoi, err = check_bool(regs['no_auto_intr_regs'],
+        no_auto_intr, err = check_bool(regs['no_auto_intr_regs'],
                                    'no_auto_intr_regs')
         if err:
             error += 1
     else:
-        no_autoi = False
-    if 'interrupt_list' in regs and 'genautoregs' not in regs and not no_autoi:
+        no_auto_intr = False
+
+    if 'no_auto_alert_regs' in regs:
+        no_auto_alerts, err = check_bool(regs['no_auto_alert_regs'],
+                                   'no_auto_alert_regs')
+        if err:
+            error += 1
+    else:
+        no_auto_alerts = False
+
+
+    if 'interrupt_list' in regs and 'genautoregs' not in regs and not no_auto_intr:
         iregs, err = make_intr_regs(regs, offset, addrsep, fullwidth)
         error += err
         autoregs.extend(iregs)
         offset += addrsep * len(iregs)
 
     # Generate a NumAlerts parameter for provided alert_list.
-    if 'alert_list' in regs:
-        num_alerts = len(regs['alert_list'])
-        # Some modules have alerts of width > 1.
-        for a in regs['alert_list']:
-            if 'width' in a:
-                a_width = int(a['width'])
-                if a_width > 1:
-                    num_alerts += (a_width - 1)
+    if regs.setdefault('alert_list', []):
+        # Generate alert test registers.
+        if 'genautoregs' not in regs and not no_auto_alerts:
+            aregs, err = make_alert_regs(regs, offset, addrsep, fullwidth)
+            error += err
+            autoregs.extend(aregs)
+            offset += addrsep * len(aregs)
+
+        num_alerts = sum([int(x['width'], 0) if 'width' in x else 1 for x in regs['alert_list']])
```

Comment:
This could be written with `get` like in [the comment above](https://github.com/lowRISC/opentitan/pull/3922/commits/57b2c5204412d4324f27d978c17da4d712ba911e#r512542381).

Also, is "num_alerts" the right name here? Does an alert register with width 8 mean 8 alerts?

Reply:
fixed.

yes, some designs use that. although it is preferred to name each alert individually, instead of declaring a bus. I'll add a warning to let the user know.